apps/web/src/app/[locale]/tracks/_components/track-card.tsx

```diff
@@ -1,56 +1,86 @@
 'use client';
 
-import clsx from 'clsx';
-import { useRouter } from 'next/navigation';
-import { TrackChallenge } from './track-challenge-card';
 import type { Tracks } from './track-grid';
-import { Button } from '@repo/ui/components/button';
+import { Card, CardContent } from '@repo/ui/components/card';
+import { Badge } from '@repo/ui/components/badge';
+import Link from 'next/link';
+import { clsx } from 'clsx';
+import { Swords } from '@repo/ui/icons';
 
 interface TrackProps {
+  index: number;
   track: Tracks[number];
 }
 
-export function TrackCard({ track }: TrackProps) {
-  const router = useRouter();
+const BGS_BY_TRACK: Record<number, string> = {
+  0: 'to-difficulty-beginner/30 dark:to-difficulty-beginner-dark/20',
+  1: 'to-difficulty-easy/30 dark:to-difficulty-easy-dark/20',
+  2: 'to-difficulty-medium/30 dark:to-difficulty-medium-dark/20',
+  3: 'to-difficulty-hard/30 dark:to-difficulty-hard-dark/20',
+  4: 'to-difficulty-extreme/30 dark:to-difficulty-extreme-dark/20',
+  5: 'to-difficulty-hard/10 dark:to-difficulty-hard-dark/10',
+  6: 'to-difficulty-medium/10 dark:to-difficulty-medium-dark/10',
+  7: 'to-difficulty-easy/10 dark:to-difficulty-easy-dark/10',
+  8: 'to-difficulty-beginner/10 dark:to-difficulty-beginner-dark/10',
+  9: 'to-difficulty-extreme/10 dark:to-difficulty-extreme-dark/10',
+} as const;
+
+const EnrolledBadge = ({ text = 'Enrolled' }: { text?: string }) => (
+  <div
+    className={clsx(
+      'absolute -right-[1px] -top-[20px] z-[-1] rounded rounded-tl-lg rounded-tr-xl px-5 pb-10 pt-[3px] text-[10px]',
+      'border border-blue-400 bg-blue-400 text-white',
+    )}
+  >
+    {text}
+  </div>
+);
+
+export function TrackCard({ track, index }: TrackProps) {
+  const isEnrolled = Boolean(track.enrolledUsers?.length);
+
   return (
-    <div className="relative w-full">
-      <div className="w-full">
-        <div
-          className={clsx(
-            'relative inset-0 flex max-w-[400px] flex-col justify-start space-y-4',
-            'max-md:scale-110',
-          )}
-        >
-          <div className="flex w-[69%] items-center justify-between gap-3 rounded-b-lg rounded-t-xl bg-neutral-500/10 p-2 pl-3">
-            <span className="flex items-center gap-1 text-sm font-semibold tracking-wide">
-              {track.title}
-            </span>
+    <Link href={`/tracks/${track.id}`} className="group">
+      <Card
+        className={clsx(
+          'relative transition-colors duration-300',
+          isEnrolled
+            ? 'border-blue-400'
+            : 'group-hover:border-blue-400 group-focus:border-blue-400',
+        )}
+      >
+        {track.enrolledUsers?.length ? <EnrolledBadge /> : null}
+        <CardContent className="flex items-center gap-5 p-2 pr-4">
+          <div
+            className={clsx(
+              `bg-gradient-to-r from-neutral-500/10 from-10% ${BGS_BY_TRACK[index]} relative to-100% dark:from-neutral-500/20`,
```

Comment:
does index actually correlate with difficulty here?

Reply:
No there is no difficulty level set for a track. Here the index denotes the iteration count of the `tracks.map`.

We will have max 10 tracks and I don't know what will be the name/slug of the track so for now I used the element index values for assigning the unique gradient.
@bautistaaa